text/0000-vhdl-emitter.md

```diff
@@ -0,0 +1,138 @@
+firrtl -E verilog -E vhdl
+```
+
+Similarly, a user can use explicit annotations (a `Seq[Annotation]` or `AnnotationSeq`) to control which emitters run. Users, both existing and new should be able to use the simple `Array[String]` or `AnnotationSeq` APIs to control which emitter runs without having to deal with any internals.
+
+This is a pure API addition to add the VHDL emitter and is not expected to change anything with other emitters. Any VHDL-specific optimizations will be relegated to only running inside the VHDL emitter transform.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Like the Verilog emitters, VHDL (or other) emitters should be entirely self-contained. This is necessary if optimizations, which may be divergent from downstream Phase/Transform needs, are required. In short, the VHDL emitter should be implemented just like the Verilog emitter in terms of architecture.
+
+The VHDL emitter should be able to be run with other emitters without impact. E.g., the Verilog emitter and the VHDL emitter should be able to be run together.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+
+## Verilog/VHDL Fungibility
+
+Verilog and VHDL are entirely fungible from an FPGA/ASIC toolflow perspective. Mixed-language (using both Verilog and VHDL) designs are supported by all major tool vendors. Consequently, for most intents and purposes, Verilog is just as good as VHDL (and vice versa) from the perspective of putting something on an FPGA or taping out a chip.
+
+This contrasts with how things work for backends of a software compiler. A software compiler backend is targeting a different ISA and expected to be different. In this case, the VHDL backend can be viewed as being redundant whereas software compiler backends are orthogonal.
+
+The fact that the Verilog and VHDL backends are fungible/redundant then has problems of maintainability and validation. For maintainability, adding the VHDL backend means that any features going into either backend needs to be available in the other. This puts more of a load on existing FIRRTL developers and will slow down our cadence. For validation, you should be able to tape-out using either the VHDL or Verilog path and get the same result. This means that an implementation of this RFC would require doing some kind of equivalence checking on the Verilog and VHDL output. Doing this with open source tools may not be possible.
+
+## Open Source Tooling
+
+VHDL has no open source compiler equivalent to [Verilator](https://github.com/verilator/verilator). Some open source tools do exist, notably [GHDL](https://github.com/ghdl/ghdl), but the performance is not up to snuff with Verilator or VCS. Integration with GHDL would likely be needed as part of this PR and that adds further complexity of needing to maintain additional back-ends in testers. The existing factoring of things where post-emission back-ends live in testers may be worth reconsidering.
+
```

Comment:
There is also [nvc](https://github.com/nickg/nvc).

Reply:
Ah, cool. I wasn't aware of this one...

Reply:
Seeing build status it does not build up much trust, but I was able to run programaticaly translated RISC-V 7months back.